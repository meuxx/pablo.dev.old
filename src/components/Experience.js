import React from 'react'

export default () => (
  <section id="two">
    <h2>Experience</h2>
    <h3>
      Kooomo (Dublin, Ireland) <strong>April 2014 â€“ Present</strong>
    </h3>
    <p>
      Developed and supported the ecommerce platform with special focus on
      performance improvements and the implementation of new technologies to
      satisfy the business needs. Creation of several internal tools to monitor
      the platform and to ensure the healthy status of it. Coordinated the
      delivery of new technology implementations alongside the Sysadmin team.
      Responsible of the code reviews and the deployments. I was also
      responsible for the training / mentoring of the new employees.
    </p>
  </section>
)
