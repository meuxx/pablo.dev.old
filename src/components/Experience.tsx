import React from 'react'
import { Section, SectionHeader, Experience as ExperienceArticle } from '../styles/styles'

interface ExperienceInterface {
  company: string
  location: string
  period: string
  description: string
}
const experienceData: ExperienceInterface[] = [
  {
    company: `Sainsbury's`,
    location: 'London, United Kingdom',
    period: 'August 2019 – Present',
    description: '',
  },
  {
    company: `Kooomo`,
    location: 'Dublin, Ireland',
    period: 'April 2014 – August 2019',
    description: `Developed and supported the ecommerce platform with special focus on performance improvements and the
    implementation of new technologies to satisfy the business needs. Creation of several internal tools to monitor the
    platform and to ensure the healthy status of it. Coordinated the delivery of new technology implementations
    alongside the Sysadmin team. During my last year in Kooomo I had the opportunity to lead the SRE team as well as a
    project to modernise the monolithic platform codebase.`,
  },
]

const Experience: React.FC = () => (
  <Section>
    <SectionHeader>
      <h2>Experience</h2>
    </SectionHeader>
    {experienceData.map(({ company, location, period, description }, index) => (
      <ExperienceArticle key={`experience-${index}`}>
        <h3>
          {company} ({location}) <strong>{period}</strong>
        </h3>
        <p>{description}</p>
      </ExperienceArticle>
    ))}
  </Section>
)

export default Experience
