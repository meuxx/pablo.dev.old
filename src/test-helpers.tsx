import { render } from '@testing-library/react'
import { Themed } from './styles/GlobalStyle'
import React from 'react'

const renderComponent: typeof render = (ui, options) => render(ui, { wrapper: Themed, ...options })

const mockComponent = (componentName: string): React.FC => (props) => {
  const { children, ...otherProps } = props
  return (
    <div data-component={componentName} data-props={JSON.stringify(otherProps, null, ' ')}>
      {children}
    </div>
  )
}

export { renderComponent, mockComponent }
