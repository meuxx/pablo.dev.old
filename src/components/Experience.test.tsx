import React from 'react'
import { renderComponent } from '../test-helpers'
import Experience from './Experience'

describe('Experience', () => {
  it('renders without crashing', () => {
    const data = [
      {
        company: 'Test company 1',
        location: 'London, United Kingdom',
        period: 'January 2019 – Present',
        description:
          'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec nec dolor leo. Interdum et malesuada fames ' +
          'ac ante ipsum primis in faucibus. Quisque vulputate volutpat nisi et auctor.',
      },
      {
        company: 'Test company 2',
        location: 'Tokyo, Japan',
        period: 'August 2008 – January 2019',
        description:
          'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla non sapien ligula. Integer dignissim ' +
          'luctus augue sit amet commodo. Nunc hendrerit interdum sapien sed pretium.',
      },
    ]

    const { container } = renderComponent(<Experience data={data} />)

    expect(container).toBeDefined()
    expect(container).toMatchSnapshot()
  })
})
