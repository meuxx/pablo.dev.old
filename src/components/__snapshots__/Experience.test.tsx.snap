// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience renders without crashing 1`] = `
.c0 {
  margin: 2em 0 0 0;
  padding: 2em 0 0 0;
}

.c0:first-child {
  margin-top: 0;
  padding-top: 0;
}

.c0 p {
  color: #b2b2b2;
  position: relative;
  margin: 0 0 2em 0;
}

.c0 p strong {
  color: #ffffff;
  font-weight: 700;
}

.c1 h1,
.c1 h2 {
  color: #ffffff;
  font-weight: 700;
}

.c1 h1 {
  margin: 0 0 1em;
  line-height: 1.5em;
}

.c1 h2 {
  margin: 0 0 1em;
  font-size: 1.5em;
  line-height: 1.5em;
}

.c2 h3 {
  color: #ffffff;
  font-weight: 700;
  font-size: 1.25em;
  line-height: 1.5em;
}

.c2 h3 strong {
  color: #0B61A4;
  font-weight: 700;
  float: right;
}

@media screen and (max-width:736px) {
  .c0 {
    margin: 1.5em 0 0 0;
    padding: 1.5em 0 0 0;
  }
}

@media screen and (max-width:736px) {
  .c1 h2 {
    font-size: 1.2em;
  }
}

@media screen and (max-width:736px) {
  .c2 h3 {
    font-size: 1em;
  }
}

<div>
  <section
    class="c0"
  >
    <header
      class="c1"
    >
      <h2>
        Experience
      </h2>
    </header>
    <article
      class="c2"
    >
      <h3>
        Test company 1
         (
        London, United Kingdom
        ) 
        <strong>
          January 2019 – Present
        </strong>
      </h3>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec nec dolor leo. Interdum et malesuada fames ac ante ipsum primis in faucibus. Quisque vulputate volutpat nisi et auctor.
      </p>
    </article>
    <article
      class="c2"
    >
      <h3>
        Test company 2
         (
        Tokyo, Japan
        ) 
        <strong>
          August 2008 – January 2019
        </strong>
      </h3>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla non sapien ligula. Integer dignissim luctus augue sit amet commodo. Nunc hendrerit interdum sapien sed pretium.
      </p>
    </article>
  </section>
</div>
`;
